File: WebFormSample/Content/JSonStringBuilder.js
function JSonStringBuilder() {
    this.builder = "";
}

JSonStringBuilder.prototype.Begin = function () {
    this.builder += "{";
}

JSonStringBuilder.prototype.End = function () {
    this.builder = this.builder.substr(0, this.builder.length - 1);
  
    this.builder += "}";
}

//不管传入的data是什么类型，都作为字符串类型处理，所以数据需要加上双引号。
JSonStringBuilder.prototype.Add = function (key, data) {
    this.builder += "\"" + key + "\":" + "\"" + data + "\",";
}

JSonStringBuilder.prototype.toString = function () {
    return this.builder;
}

File: WebFormSample/Lib/JSonStringBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

using System.Text;

namespace WebFormSample.Lib
{
    public class JSonStringBuilder
    {
        private StringBuilder builder;

        public JSonStringBuilder()
        {
            builder = new StringBuilder();
        }

        public void Begin()
        {
            builder.Append("{");
        }

        public void End()
        {
            builder.Remove(builder.Length - 1, 1);
            builder.Append("}");
        }

        public void Add(string key, object data)
        {
            if (data.GetType() == typeof(String))
            {
                builder.AppendFormat("\"{0}\":\"{1}\",", key, data);
            }
            else
            {
                builder.AppendFormat("\"{0}\":{1},", key, data);
            }
        }

        public void Add(string josnStr)
        {
            builder.Append(josnStr+",");
        }

        public override string ToString()
        {
            string str = builder.ToString();
            return str;
        }
    }
}